validator/sawtooth_validator/execution/tp_state_handlers.py

```diff
@@ -98,3 +98,34 @@ def handle(self, connection_id, message_content):
             HandlerStatus.RETURN,
             response,
             validator_pb2.Message.TP_STATE_SET_RESPONSE)
+
+
+class TpStateDeleteHandler(Handler):
+
+    def __init__(self, context_manager):
+        self._context_manager = context_manager
+
+    def handle(self, connection_id, message_content):
+        delete_request = state_context_pb2.TpStateDeleteRequest()
+        delete_request.ParseFromString(message_content)
+        try:
+            return_values = self._context_manager.delete(
+                delete_request.context_id, list(delete_request.addresses))
+        except AuthorizationException:
+            response = \
+                state_context_pb2.TpStateDeleteResponse(
+                    status=state_context_pb2.
+                    TpStateDeleteResponse.AUTHORIZATION_ERROR)
+            return HandlerResult(
+                HandlerStatus.RETURN,
+                response,
+                validator_pb2.Message.TP_STATE_DELETE_RESPONSE)
+
+        LOGGER.debug("DELETE: %s", return_values)
+        response = state_context_pb2.TpStateDeleteResponse(
+            addresses=return_values,
```

Comment:
This will throw a `TypeError`. `addresses` should be a list of strings, not the boolean returned by `delete` on line 112. I do think a boolean probably makes more sense than a list of addresses, but it needs to be either one or the other.